### PR TITLE
AR representation: speed improvement and clean-ups

### PR DESCRIPTION
**Pull request by Sabrina85**

### CODE REVIEW

File: src/odemis/util/angleres.py [hunk 1]
```diff
@@ -87,76 +84,114 @@ def _extractAngleInformation(data, hole, dtype=None):
     # offset of 2 px should be sufficient for all image sizes and
```
Comment (pieleric) [inline src/odemis/util/angleres.py]: Please also update the comment saying that 2px is enough all time ;-)

File: src/odemis/util/angleres.py [hunk 2]
```diff
@@ -87,76 +84,114 @@ def _extractAngleInformation(data, hole, dtype=None):
...
 
     return theta_data, phi_data, intensity_data, circle_mask_dilated
 
 
-def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
+def _FindAngle(x_array, y_array, pixel_size, parabola_f):
+    """
+    For given pixels, finds the angle of the corresponding ray
+    :parameter x_array: (ndarray) x coordinates of the pixels
+    :parameter y_array: (ndarray) y coordinate of the pixel
```
Comment (pieleric) [inline src/odemis/util/angleres.py]: Please put a "s" in the words also for Y, unless there is really a difference?

File: src/odemis/util/angleres.py [hunk 3]
```diff
@@ -87,76 +84,114 @@ def _extractAngleInformation(data, hole, dtype=None):
...
+    :parameter pixel_size: (float, float) detector pixel size (X/Y)
+    :parameter parabola_f: (float) parabola_parameter=1/(4f): f: focal point of mirror (place of sample)
+    :returns: (ndarray) theta, phi (the corresponding spherical coordinates for each pixel in detector)
+                              and omega (solid angle: angular range collected per px)
+    """
+
+    y = x_array * pixel_size[0]
```
Comment (pieleric) [inline src/odemis/util/angleres.py]: What's the idea behind naming the X data `y`, and the Y data `z`? Maybe if you were using entirely different letters, it'd be less confusing. Or add a comment explaining.

File: src/odemis/util/angleres.py [hunk 4]
```diff
@@ -87,76 +84,114 @@ def _extractAngleInformation(data, hole, dtype=None):
...
+    # theta
+    theta = numpy.arccos(z / sqrtxfocus2plusr2)
+
+    # phi
+    # negative xfocus to put phi0 and phi180 to match the raw data (convention we chose for display ar data)
+    # also map values from -pi to pi -> 0 to 2pi
+    phi = numpy.arctan2(y, -xfocus) % (2 * math.pi)
+
+    # omega
+    # omega = (pixel_size[0] * pixel_size[1]) * \
```
Comment (pieleric) [inline src/odemis/util/angleres.py]: Drop this commented line.

File: src/odemis/util/angleres.py [hunk 5]
```diff
@@ -87,76 +84,114 @@ def _extractAngleInformation(data, hole, dtype=None):
...
+    intensity_data = cropped_image / omega
 
     return theta_data, phi_data, intensity_data, circle_mask_dilated
 
 
-def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
+def _FindAngle(x_array, y_array, pixel_size, parabola_f):
+    """
+    For given pixels, finds the angle of the corresponding ray
+    :parameter x_array: (ndarray) x coordinates of the pixels
```
Comment (pieleric) [inline src/odemis/util/angleres.py]: Please add the expected dimension of the ndarrays. They are all a 2D array, right?

File: src/odemis/util/angleres.py [hunk 6]
```diff
@@ -169,18 +204,24 @@ def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
...
+    # create grid of positions for interpolation: neg to pos as x/y data polar
+    # contain now values from -output_size/2 to +output_size/2
+    xi, yi = numpy.meshgrid(numpy.linspace(-output_size/2, output_size/2, output_size),
+                            numpy.linspace(-output_size/2, output_size/2, output_size))
+
     # interpolate
     qz = interp(xi, yi)
-    qz = qz.swapaxes(0, 1)[:, ::-1]  # rotate by 90°
+    # polar coordinate transformation starts with 0 at horizontal axis by definition
+    qz = numpy.rot90(qz)  # rotate by 90 degrees CCW so we start 0 at top (angles will be CW orientated)
```
Comment (pieleric) [inline src/odemis/util/angleres.py]: I didn't know there is such an explicit function! Good catch ;-)

File: src/odemis/util/angleres.py (hunk 3 above)
Comment (Sabrina85) [reply src/odemis/util/angleres.py]: yes that also gives me still headaches. I know that it had something to do where the axes regarding the mirror are defined, but I am not sure anymore which one was what. I renamed them now and put a note there. I will ask Toon when I am back and will update it then.

**Pull request merged**